src/main/java/com/shapesecurity/salvation/Parser.java

```diff
@@ -146,6 +146,8 @@ private boolean eat(@Nonnull Class<? extends Token> c) {
                 continue;
             try {
                 policy.addDirective(this.parseDirective());
```

Comment:
We just need to check here before adding the directive that the policy doesn't already contain a directive of that type.